src/libmongoc/tests/mock_server/mock-server.c

```diff
@@ -935,13 +1002,71 @@ _mock_server_receives_msg (mock_server_t *server, uint32_t flags, ...)
  */
 
 request_t *
-mock_server_receives_ismaster (mock_server_t *server)
+mock_server_receives_legacy_hello (mock_server_t *server,
```

Comment:
I changed the semantics of this function a bit: it now expects either `hello` or legacy hello commands, while the new `mock_server_receives_hello` function always expects a `hello` command.